### DISCUSSION TITLE
Iterate over functions in `tar_map()` without metaprogramming

### DISCUSSION THREAD

**Discussion by alexpghayes** [ropensci/targets#650] [General]
This is more of a design question than anything else, but I am wondering about the need for metaprogramming in `tar_map()`. For example, consider this example from Chapter 12 of the docs.

```
# _targets.R file:
library(targets)
library(tarchetypes)
library(tibble)
values <- tibble(
  method_function = rlang::syms(c("method1", "method2")),
  data_source = c("NIH", "NIAID")
)
targets <- tar_map(
  values = values,
  tar_target(analysis, method_function(data_source, reps = 10)),
  tar_target(summary, summarize_analysis(analysis, data_source))
)
list(targets)
```

Is there some reason why this is preferable to 

```
...
values <- tibble(
  method_function = list(method1, method2),  # method1 and method2 are both functions
  data_source = list(NIH, NIAID), # both data frames
)
...
```

**Comment by wlandau** [comment]
The job of `tar_map()` is to create R expressions that act as target commands, so it is hard to avoid a certain degree of metaprogramming. Also, it is preferable to work on on the actual values in the tibble instead of trying to quote the names you type, since these data frames are sometimes large and it is helpful to have some degree of automation.
